src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -2,66 +2,83 @@
 
 import org.skypro.skyshop.product.Product;
 
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+
 public class ProductBasket {
-    private final Product [] products;
-    private int size;
+    private List<Product> products;
 
     public ProductBasket() {
-        products = new Product[5];
+        products = new LinkedList<>();
     }
 
-    public void addProduct (Product newproduct) {
-        if (size >= products.length) {
-            System.out.println("Невозможно добавить продукт");
-            return;
-        }
-        products [size++] = newproduct;
+    public void addProduct(Product newproduct) {
+        products.add(newproduct);
     }
 
-    public int countPriceBasket () {
-        if (products[0] == null) {
-            return 0;
-        }
+    public int countPriceBasket() {
         int price = 0;
-        for (int i=0; i<size; i++) {
-            price += products[i].getPrice();
+        for (Product item : products) {
+            price += item.getPrice();
         }
         return price;
     }
 
-    public void printBasket () {
+    public void printBasket() {
         System.out.println();
         System.out.println("Список продуктов в корзине:");
-        if (products[0] == null) {
+        if (products.isEmpty()) {
             System.out.println("В корзине пусто");
             return;
         }
         int n = 0;
-        for (int i=0; i<size; i++) {
-            System.out.println(products[i]);
-            if (products[i].isSpecial()) {
+        for (Product item : products) {
+            System.out.println(item);
+            if (item.isSpecial()) {
                 n++;
             }
         }
-        System.out.println("Итого: "+ countPriceBasket() +" р.");
-        System.out.println("Специальных товаров: "+ n);
+        System.out.println("Итого: " + countPriceBasket() + " р.");
+        System.out.println("Специальных товаров: " + n);
     }
 
-    public boolean findProduct (String product) {
-        if (products[0] == null) {
+    public boolean findProduct(String product) {
+        if (products.isEmpty()) {
             return false;
         }
-        for (int i = 0; i < size; i++) {
-            if (product.equals(products[i].getName())) {
+        for (Product item : products) {
+            if (product.equals(item.getName())) {
                 return true;
             }
         }
         return false;
     }
 
-    public void emptyBasket () {
-        for (int i=0; i<size; i++) {
-            products[i] = null;
+    public void emptyBasket() {
+        products.clear();
+    }
+
+    public List deleteProduct(String productName) {
```

Comment:
Листы должны быть с параметром. Использование сырых типов не очень хорошая и небезопасная практики
```suggestion
    public List<Product> deleteProduct(String productName) {
```